TryGhost/Ghost
# Ember-cli, Ember, & Ember Data 1.13.x upgrades

closes #5630
- upgrade ember-cli to latest version
- upgrade ember to latest 1.13.x release
- upgrade ember data to latest 1.13.x release
  - update custom adapters and serialisers for new internal JSON-API compatible formats ([docs](http://emberjs.com/blog/2015/06/18/ember-data-1-13-released.html#toc_transition-to-the-new-jsonserializer-and-restserializer-apis))
  - update all store queries to use new standardised query methods ([docs](http://emberjs.com/blog/2015/06/18/ember-data-1-13-released.html#toc_simplified-find-methods))
  - add `ember-data-filter` addon ready for `store.filter` removal in ember-data 2.0 ([docs](http://emberjs.com/blog/2015/06/18/ember-data-1-13-released.html#toc_ds-store-filter-moved-to-an-addon))
- remove use of prototype extensions for computed properties and observers
- consolidate pagination into a single route mixin and simplify configuration

---
- [x] upgrade ember-cli to 1.13.8
- [x] upgrade ember to 1.13.10
- [x] upgrade ember-data to 1.13.12
- [x] fix `gh-navitem-url-input` component (may not be necessary after fix for #5841 is implemented)

This should fix all deprecation notices apart from the `"`lookup` was called on a Registry"` notices caused by `ember-simple-auth`. If during testing you notice any other deprecation notices in the console please comment on this PR.

`ember-simple-auth` does have an Ember 2.0 compatible branch that is not yet released as stable, the upgrade to ESA 1.0 will be handled separately as part of the move to Ember 2.0.

I believe @jaswilli's blocker that [he mentioned](https://github.com/TryGhost/Ghost/issues/5630#issuecomment-131287004) regarding model state transitions may still be an issue but no longer manifests itself due to more in-depth client-side validations. Some more testing on that front would be useful.
